### PR TITLE
Fix blob sink tests

### PR DESCRIPTION
**Pull request by pakrym**
@victorhurdugaci

### CODE REVIEW

File: src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/AzureBlobSink.cs [hunk 1]
```diff
@@ -70,6 +70,13 @@ public class AzureBlobSink : PeriodicBatchingSink
         /// <inheritdoc />
         protected override async Task EmitBatchAsync(IEnumerable<LogEvent> events)
         {
+            await EmitBatchInternalAsync(events);
+        }
+
+
+        // For unit testing
+        internal async Task EmitBatchInternalAsync(IEnumerable<LogEvent> events)
```
Comment (moozzyk) [inline src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/AzureBlobSink.cs]: you could make EmitBatchAsync `protected internal`. Potentially you could avoid InternalsVisibleTo if you derived from AzureBlobSink in your tests and created a public EmitBatchAsyncPublich method that forwards to EmitBatchAsync...

**Comment by victorhurdugaci** [comment]
What was the problem? Please add a few words when sending a PR

**Comment by pakrym** [comment]
@victorhurdugaci problem was in timer inside `PeriodicBatchingSink` firing randomly producing inconsistent batches.

File: src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/AzureBlobSink.cs [hunk 2]
```diff
@@ -70,6 +70,13 @@ public class AzureBlobSink : PeriodicBatchingSink
         /// <inheritdoc />
         protected override async Task EmitBatchAsync(IEnumerable<LogEvent> events)
         {
+            await EmitBatchInternalAsync(events);
+        }
+
```
Comment (cesarbs) [inline src/Microsoft.Extensions.Logging.AzureWebAppDiagnostics/Internal/AzureBlobSink.cs]: nit: remove extra blank line

**Comment by victorhurdugaci** [comment]
> @victorhurdugaci problem was in timer inside PeriodicBatchingSink firing randomly producing inconsistent batches.

Will this be an issue in production?

**Comment by pakrym** [comment]
@victorhurdugaci we don't really care how exact batches would look like, only that most of them would have >1 message. Yes, no longer testing a base class.

**Comment by pakrym** [comment]
:up::date:

**Pull request merged**